federation/schema.go

```diff
@@ -83,6 +92,44 @@ func convertVersionedSchemas(schemas serviceSchemas) (*SchemaWithFederationInfo,
 	fieldInfos := make(map[*graphql.Field]*FieldInfo)
 	for _, service := range serviceNames {
 		for _, typ := range serviceSchemasByName[service].Schema.Types {
+
+			// For federated fields parse the arguments to figure out which
+			// fields are the federated keys. They annotate that information
+			// on the field object.
+			if typ.Name == "Federation" {
+				for _, field := range typ.Fields {
+				    // Extract the type name from the formatting <object>-<service>
+				    // And check that the object type exists
+					names := strings.SplitN(field.Name, "-", 2)
+					if len(names) != 2 {
+						return nil, oops.Errorf("Field %s doesnt have an object name and service name", field.Name)
+					}
+					objName := names[0]
+					obj, ok := types[objName].(*graphql.Object)
+					if !ok {
+						return nil, oops.Errorf("Expected objectName %s on merged schema", objName)
+					}
+					for _, arg := range field.Args {
+						rootType := getRootType(arg.Type)
```

Comment:
Do we think we could force the rootType to be a subset of obj here?

Reply:
I added a check to ensure this!